wrangler/src/@utils/auth.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Helper functions for managing authentication against a Turret
+ */
+import BigNumber from 'bignumber.js'
+import moment from 'moment'
+import { Transaction, Networks } from 'stellar-base'
+import { verifyTxSignedBy } from './stellar-sdk-utils'
+
+/**
+ * Determine the authenticated user and contracts with the provided token.
+ * 
+ * @param {string} network The Stellar Network to validate the token on
+ * @param {string} authToken - The provided auth token
+ * @param {string?} dataKey - (Optional) Fetch all ManageData operations such that the key matches this dataKey.
+ *                           All ManageData values whose keys match will be included in the data output array.
+ * @link https://tyvdh.github.io/stellar-tss/#section/Authentication
+ * 
+ *
+ * @typedef {Object} AuthResult
+ * @property {string} publicKey - The authenticated public key from the token.
+ * @property {string[]} data - The data collected. Returns an empty array if no matching ManageData values are 
+ *                             found or no dataKey was provided.
+ *
+ * @returns {AuthResult} - The result of the authentication
+ */
+export function authTxToken(network, authToken, dataKey) {
```

Comment:
I'm not a strong JS developer.

If anybody knows a good way to manage these ENV variables out to utility functions, I would appreciate it!

Reply:
This seems like a good start to me. Eventually we need to go back through the entire app and TypeScript it all and scaffold it all better. Now that our final form is much more defined it would make sense. For another PR though methinks.